Networking/WebSockets/BuiltInWebSocket.cc

```diff
@@ -435,6 +447,16 @@ namespace litecore { namespace websocket {
                     closeWithError(_socket->error());
                 return;
             }
+            
+            // CBL-1633: Need to check for EOF here, or _socket->write returns
+            // 0 endlessly
+            if(n == 0 && _socket->atWriteEOF()) {
+                if(_socket->atReadEOF()) {
+                    closeSocket();
+                }
```

Comment:
Does it need to call close() if atReadEOF() returns false like it has been done in readFromSocket()?

Reply:
I don't think so because close will try to write, but we are already at writeEOF so we can't write anymore.

Reply:
Got it.